### DISCUSSION TITLE
Restricting process count to 1

### DISCUSSION THREAD

**Discussion by Mag1cByt3s** [outline/outline#5957] [Self Hosting] [answered]
Note: Restricting process count to 1 due to use of collaborative service

I have this message in my docker logs.

I have set the WEB_CONCURRENCY variable to 24 so outline can spawn 24 processes for better performance. However it is always restricting the process count to 1 for some reason.

How can I fix this?

**Comment by alkanna** [comment]
I saw this piece of code :

```
if (serviceNames.includes("collaboration")) {
  if (processCount !== 1) {
    Logger.info(
      "lifecycle",
      "Note: Restricting process count to 1 due to use of collaborative service"
    );
  }

  processCount = 1;
}
```

I gather from this that you must run the Collaboration service on a separate container, probably because it does not handle concurrent usage. Right now the provided docker image bundles everything : `collaboration,websockets,worker,web`

**Reply by Mag1cByt3s** [reply]
I see, thanks for the insight.

I am not even sure if i need the collaboration feature since i am the only user on my server.

Is there a way to disable collaboration?

**Reply by tommoor** [reply, marked as answer]
If you're only one user you probably don't need 24 process concurrency either – there are servers with thousands of users running on a single process 😉 

There is no way to disable collaborative editing, it's core to how the syncing of changes from client <-> server works.

**Reply by ovizii** [reply]
Just out of curiosity, and despite not understanding why a single user would need 24 processes, I'm left wondering why we have an option called WEB_CONCURRENCY when as you say:

> there is no way to disable collaborative editing, it's core to how the syncing of changes from client <-> server works. 

What purpose does an option serve which is always override by the design of outline?

**Reply by alkanna** [reply]
Because you can scale other services except collaboration, this option does not include it as shown in the piece of code above.

**Reply by ovizii** [reply]
Thanks, digging around I found docs/SERVICES.md) and now it all makes more sense.
